error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/StreamRules.java

```diff
@@ -399,6 +412,20 @@ int after(Stream<T> stream, ToIntFunction<T> mapper) {
     }
   }
 
+  static final class StreamCollectSummingInt<T> {
+    @BeforeTemplate
+    long before(
+        Stream<T> stream,
+        @Matches(IsLambdaExpressionOrMethodReference.class) ToIntFunction<T> mapper) {
```

Comment:
I could not Generalise `Function<? super T, Integer> mapper` from `StreamMapToIntSum`....

Reply:
Thank you so much for the clarification. 🚀 

One question though
should'nt 
```
....Function<? super T, Integer> mapper) {
  return stream.collect(summingInt(t -> mapper.apply(t).intValue()));
}
```

be equal to 

```
....ToIntFunction<T> mapper) {
 return stream.collect(summingInt(mapper));
}
```
The tests fail but the compiler aint complaining 😓 ...